extensions/smallrye-metrics/runtime/src/main/java/io/quarkus/smallrye/metrics/runtime/SmallRyeMetricsFactory.java

```diff
@@ -0,0 +1,203 @@
+package io.quarkus.smallrye.metrics.runtime;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+import java.util.function.ToDoubleFunction;
+
+import org.eclipse.microprofile.metrics.Counter;
+import org.eclipse.microprofile.metrics.Gauge;
+import org.eclipse.microprofile.metrics.MetricRegistry;
+import org.eclipse.microprofile.metrics.MetricType;
+import org.eclipse.microprofile.metrics.Tag;
+import org.eclipse.microprofile.metrics.Timer;
+
+import io.quarkus.runtime.metrics.MetricsFactory;
+import io.smallrye.metrics.MetricRegistries;
+
+public class SmallRyeMetricsFactory implements MetricsFactory {
+    MetricRegistry registry = MetricRegistries.get(MetricRegistry.Type.VENDOR);
```

Comment:
Some metrics created by extensions go to BASE, so this needs to be a changeable

Reply:
this is hard to translate as the notion of base or vendor doesn't apply.. but I will note it as optional/ignored in some contexts

Reply:
This is smallrye-specific so here this notion does apply (at least for the time being), doesn't it? Could we make the registry type a field of this class and set it by a constructor parameter?

Reply:
Yes, this is what I was thinking. Would be a simple enum on this interface, rather than pulling in MP Metrics dependency.